### PR TITLE
feat: add cloud images to releases

### PR DESCRIPTION
**Pull request by rsmitty**
This PR updates our CI so that when we release talos, a json file
containing our cloud images for AWS will be published as a release
asset.

Signed-off-by: Spencer Smith <robertspencersmith@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/3820)

### CODE REVIEW

File: .drone.jsonnet [hunk 1]
```diff
@@ -488,29 +488,12 @@ local conformance_pipelines = [
...
-  },
-};
-
-local upload_images_pipeline = Pipeline('upload-images', upload_images_steps) + upload_images_trigger;
-
 // Release pipeline.
 
 local boot = Step('boot', depends_on=[e2e_docker, e2e_qemu]);
 
+local cloud_images = Step("cloud-images", depends_on=[load_artifacts], environment=creds_env_vars);
```
Comment (smira) [inline .drone.jsonnet:495]: I think `depends_on` should be updated with same dependencies as steps around, `e2e_docker, e2e_qemu`

Comment (rsmitty) [reply .drone.jsonnet]: yep, good call. fixed in amended commit.

**Review by smira — APPROVED** [review]

**Pull request merged**